src/platform/plugins/shared/dashboard/public/dashboard_renderer/dashboard_renderer.tsx

```diff
@@ -59,6 +61,12 @@ export function DashboardRenderer({
     if (dashboardApi) dashboardApi.locator = locator;
   }, [dashboardApi, locator]);
 
+  useEffect(() => {
+    if (dashboardInternalApi) {
+      dashboardInternalApi.setDashboardContainerRef(dashboardContainerRef.current);
+    }
+  }, [dashboardInternalApi, dashboardContainerRef]);
```

Comment:
refs do not trigger re-renders so lets remove dashboardContainerRef from this useEffect dependency array.

dashboardContainerRef$ may already be this value if `ref` callback fires first. So how about checking equality before calling set.

Reply:
Removes in https://github.com/elastic/kibana/pull/233226/commits/c9ef5db458d78b71e3b6ac1fa9cd9855186d90e0